dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/util/BufferedWriter.java

```diff
@@ -121,7 +122,14 @@ public BufferedWriter(int size)
         { // 306998.15
             Tr.debug(tc, "BufferedWriter(), size --> " + size + " this --> " + this);
         }
-        buf = new char[size];
+
+        if (size <= 0){
+            buf = EMPTY_CHAR_ARRAY;
+        }
+        else{
+            buf = new char[size];
+        }
+
         bufferSize = size;
```

Comment:
You are checking for < 0.  Before that would result in an exception which maybe is a good thing.  Maybe all the code could just be:

buf = (size == 0 ? EMPTY_CHAR_ARRAY : new char[size]);

That way you get the old behavior for negative sizes passed in resulting in an exception.